specification/dartLangSpec.tex

```diff
@@ -1317,127 +1350,256 @@ \section{Variables}
 A mutable variable declaration of the form
-\code{{} $T$ $v$;}
-or \code{$T$ $v$ = $e$;}
-induces an implicit setter function (\ref{setters}) with signature
-\code{\VOID{} \SET{} $v$=($T$ $x$)}
-whose execution sets the value of $v$ to the incoming argument $x$.
+\code{\STATIC?\,\,\LATE?\,\,\VAR\,\,\id{} = $e$;}
+implicitly induces a setter with the header
+\code{\VOID\,\,\SET\,\,\id(\DYNAMIC\,\,$x$)},
+whose execution sets the value of \id{} to the incoming argument $x$.
+
+\commentary{%
+Type inference could have provided a type different from \DYNAMIC{}
+(\ref{overview}).%
+}
+\EndCase
 
 \LMHash{}%
+\Case{Setter: Mutable variable with no declared type, no initialization}
 A mutable variable declaration of the form
-\code{\VAR{} $v$;}
-or \code{\VAR{} $v$ = $e$;}
-induces an implicit setter function with signature
-\code{\VOID{} \SET{} $v$=(\DYNAMIC{} $x$)}
-whose execution sets the value of $v$ to the incoming argument $x$.
+\code{\STATIC?\,\,\LATE?\,\,\VAR\,\,\id;}
+implicitly induces a setter with the header
+\code{\VOID\,\,\SET\,\,\id(\DYNAMIC\,\,$x$)},
+whose execution sets the value of \id{} to the incoming argument $x$.
+
+\commentary{%
+Type inference has not yet been specified in this document
+(\ref{overview}).
+Note that type inference could change, e.g.,
+\code{\VAR\,\,x;} to \code{$T$\,\,x;},
+which would take us to an earlier case.%
+}
+\EndCase
+
+\LMHash{}%
+\Case{Setter: Late-final variable with declared type}
+A variable declaration of the form
+\code{\STATIC?\,\,\LATE\,\,\FINAL\,\,$T$\,\,\id;}
+implicitly induces a setter (\ref{setters}) with the header
+\code{\VOID\,\,\SET\,\,\id($T$\,\,$x$)}.
+If this setter is executed
+in a situation where the variable \id{} has not been bound,
+it will bind \id{} to the object that $x$ is bound to.
+If this setter is executed
+in a situation where the variable \id{} has been bound to an object,
+a dynamic error occurs.
+\EndCase
+
+\LMHash{}%
+\Case{Setter: Late-final variable with no declared type, no initialization}
+A variable declaration of the form
+\code{\STATIC?\,\,\LATE\,\,\FINAL\,\,\id;}
+implicitly induces a setter with the header
+\code{\VOID\,\,\SET\,\,\id(\DYNAMIC\,\,$x$)}.
+An execution of said setter
+in a situation where the variable \id{} has not been bound
+will bind \id{} to the object that the argument $x$ is bound to.
+An execution of the setter
+in a situation where the variable \id{} has been bound to an object
+will incur a dynamic error.
+\EndCase
 
 \LMHash{}%
 The scope into which the implicit getters and setters are introduced
 depends on the kind of variable declaration involved.
 
 \LMHash{}%
 A library variable introduces a getter into
-the top level scope of the enclosing library.
+the library scope of the enclosing library.
 A class variable introduces a static getter into
-the immediately enclosing class.
+%% TODO(eernst): Come enhanced enums, add enum to this list.
+the body scope of the immediately enclosing class, mixin, or extension.
```

Comment:
This has come to pass, so is this in anticipation of including that in the spec?

Maybe just say `or enum` today. It's not wrong when you can't declare them, and then we won't forget when we do allow it.

Should it be "body scope of .... extension declaration."
(Do we even have a notion of a "class" concept separate from the declaration? Formally?)